src/coordinator/api/v1/handler/database/create.go

```diff
@@ -0,0 +1,249 @@
+		handler.Error(w, rErr.Error(), rErr.Code())
+		return
+	}
+
+	nsRegistry, err := h.namespaceAddHandler.Add(namespaceRequest)
+	if err != nil {
+		logger.Error("unable to add namespace", zap.Any("error", err))
+		handler.Error(w, err, http.StatusInternalServerError)
+		return
+	}
+
+	initPlacement, err := h.placementInitHandler.Init(placementRequest)
+	if err != nil {
+		// Attempt to delete the namespace that was just created to maintain idempotency
+		err = h.namespaceDeleteHandler.Delete(namespaceRequest.Name)
+		if err != nil {
+			logger.Error("unable to delete namespace we just added", zap.Any("error", err))
+			handler.Error(w, err, http.StatusInternalServerError)
+			return
+		}
+
+		logger.Error("unable to add namespace", zap.Any("error", err))
+		handler.Error(w, err, http.StatusInternalServerError)
+		return
+	}
+
+	placementProto, err := initPlacement.Proto()
+	if err != nil {
+		logger.Error("unable to get placement protobuf", zap.Any("error", err))
+		handler.Error(w, err, http.StatusInternalServerError)
+		return
+	}
+
+	resp := &admin.DatabaseCreateResponse{
+		Namespace: &admin.NamespaceGetResponse{
+			Registry: &nsRegistry,
+		},
+		Placement: &admin.PlacementGetResponse{
+			Placement: placementProto,
+		},
+	}
+
+	handler.WriteProtoMsgJSONResponse(w, resp, logger)
+}
+
+func (h *createHandler) parseRequest(r *http.Request) (*admin.NamespaceAddRequest, *admin.PlacementInitRequest, *handler.ParseError) {
+	dbCreateReq := new(admin.DatabaseCreateRequest)
+	if err := jsonpb.Unmarshal(r.Body, dbCreateReq); err != nil {
+		return nil, nil, handler.NewParseError(err, http.StatusBadRequest)
+	}
+
+	if dbCreateReq.Type != dbTypeLocal {
+		return nil, nil, handler.NewParseError(errInvalidDBType, http.StatusBadRequest)
+	}
+
+	if util.HasEmptyString(dbCreateReq.NamespaceName, dbCreateReq.Type) {
+		return nil, nil, handler.NewParseError(errMissingRequiredField, http.StatusBadRequest)
+	}
+
+	namespaceAddRequest := defaultedNamespaceAddRequest(dbCreateReq)
+	placementInitRequest := defaultedPlacementInitRequest(dbCreateReq)
+
+	return namespaceAddRequest, placementInitRequest, nil
+}
+
+func defaultedNamespaceAddRequest(r *admin.DatabaseCreateRequest) *admin.NamespaceAddRequest {
+	options := dbnamespace.NewOptions()
+
+	switch r.Type {
+	case dbTypeLocal:
+		options.SetBootstrapEnabled(true)
+		options.SetFlushEnabled(true)
+		options.SetWritesToCommitLog(true)
+		options.SetCleanupEnabled(true)
+		options.SetRepairEnabled(false)
+		options.SetSnapshotEnabled(false)
+
+		// ExpectedSeriesDatapointsPerHour takes precedence over RetentionPeriodNanos
+		if r.ExpectedSeriesDatapointsPerHour > 0 {
+			// We want around 720 datapoints per block, so:
+			// (720 / r.ExpectedSeriesDatapointsPerHour) * 60 * 60000000000
+			options.RetentionOptions().SetRetentionPeriod(time.Duration(2592000000000000 / r.ExpectedSeriesDatapointsPerHour))
+		} else {
+			options.RetentionOptions().SetRetentionPeriod(time.Duration(r.RetentionPeriodNanos))
+		}
+
```

Comment:
Any suggestions on what values? Do you think a better place for this would be at the options setter function level instead of here? Otherwise we'd have to duplicate this elsewhere as well.

Reply:
re: values - for a single node deploy it should probably be in the realm of min 30m and max 2h. 

Re: where this belongs - i think this code has the most context about how the cluster is being used. How about we start by putting these limits here and see how it evolves over time?

Reply:
Sounds good!